src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -439,6 +444,23 @@ public function setCheckLoginOnServer($bool)
         $this->checkLoginOnServer = $bool;
     }
 
+    public function setCaseSensitiveEncoding($encoding)
+    {
+        $this->caseSensitiveEncoding = $encoding;
+    }
+
+    private function getCaseSensitiveEncoding()
+    {
+        if (!$this->caseSensitiveEncoding) {
+            $params = $this->conn->getParams();
+            $charset = isset($params['charset']) ? $params['charset'] : 'utf8';
+
+            return $charset === 'binary' ? $charset : $charset . '_bin';
```

Comment:
please do `$this->caseSensitiveEncoding = $charset === 'binary' ? $charset : $charset.'_bin';` instead of return, so its only done once